drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -1180,7 +1181,7 @@ static bool rswitch_rx_chain(struct net_device *ndev, int *quota, struct rswitch
 			skb->dev = ndev;
 		}
 
-		if (learn_chain) {
+		if (learn_chain && rdev->priv->offload_enabled) {
```

Comment:
Here should be separate if() with read_lock I think

Reply:
Thank you for your review.
Actually, this check was added to simplify skipping not needed steps and works slowly but properly without it.
It works fine because rswitch_disable_offload already has all needed locks and after it unlocks, lists with routes for every device become empty. As a result rswitch_add_ipv4_forward doesn't do anything in this case even if offload_enabled flag is true. So, I think, there are no needings to add locks here to protect offload_enabled flag.